host/src/CoreToolsHost/LocalSettingsJsonParser.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace CoreToolsHost
+{
+    internal static class LocalSettingsJsonParser
+    {
+        internal static async Task<JsonDocument> GetLocalSettingsJsonAsJObjectAsync()
+        {
+            var fullPath = Path.Combine(Environment.CurrentDirectory, "local.settings.json");
+
+            if (File.Exists(fullPath))
+            {
+                string fileContent = "";
+                using (var fileStream = File.Open(fullPath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite | FileShare.Delete))
+                using (var streamReader = new StreamReader(fileStream))
+                {
+                    fileContent = await streamReader.ReadToEndAsync();
+                }
+                if (!string.IsNullOrEmpty(fileContent))
+                {
+                    var localSettingsJObject = JsonDocument.Parse(fileContent);
```

Comment:
this line gave me a bad weekend :(  glad #3866 is addressing some of it (but still not allowing my sweet sweet trailing commas :D